[tests][reducer] intermittent test_idle_timeout failure

This was seen in CI on a Windows machine.
```
______________________________ test_idle_timeout ______________________________

monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x00000000077C2D48>
tmp_path = WindowsPath('c:/users/travis/appdata/local/temp/pytest-of-travis/pytest-0/test_idle_timeout0')

    def test_idle_timeout(monkeypatch, tmp_path):
        "idle is not polled until idle_timeout has passed, and idle_poll results in timeout"
        now = [0]
        first_poll = []
    
        def mytime():
            now[0] += 1
            return now[0]
    
        class MyTarget(FakeTarget):
    
            def poll_for_idle(self, *args, **kwds):
                if not first_poll:
                    first_poll.append(now[0])
                FakeTarget.poll_for_idle(self, *args, **kwds)
                return time.time() >= 45
    
        class MyServer(FakeServer):
    
            def serve_testcase(self, *args, **kwds):
                continue_cb = kwds["continue_cb"]
                while continue_cb():
                    pass
                return sapphire.SERVED_ALL, []
    
        monkeypatch.setattr(time, "time", mytime)
        monkeypatch.setattr(sapphire, "Sapphire", MyServer)
    
        obj = Interesting([], MyTarget(), 30, False, False, 0, 1, 1, 10, 25, 30, FakeReduceStatus(), False)
        create_target_binary(obj.target, tmp_path)
        (tmp_path / "test.html").touch()
        obj.reduce_file = str(tmp_path / "test.html")
        obj.init(None)
        (tmp_path / "lithium0").mkdir()
        assert obj.interesting(None, tmp_path / "lithium0")
>       assert abs(first_poll[0] - 30) <= 5, "polling started at %d" % (first_poll[0],)
E       IndexError: list index out of range

grizzly\reduce\test_interesting.py:470: IndexError
```

--- a/grizzly/reduce/interesting.py
+++ b/grizzly/reduce/interesting.py
@@ -139,8 +139,9 @@ class Interesting(object):
         updated.append('suppressions=\'%s\'' % supp_file)
         self.env_mod[opt_key] = ':'.join(updated)
 
-    def monitor_process(self, iteration_done_event, idle_timeout_event):
+    def monitor_process(self, iteration_done_event, idle_timeout_event, monitor_launched):
         # Wait until timeout is hit before polling
+        monitor_launched.set()
         LOG.debug('Waiting %r before polling', self.idle_timeout)
         exp_time = time.time() + self.idle_timeout
         while exp_time >= time.time() and not iteration_done_event.is_set():
@@ -347,9 +348,11 @@ class Interesting(object):
             idle_timeout_event = threading.Event()
             iteration_done_event = threading.Event()
             if self.idle_poll:
+                monitor_launched = threading.Event()
                 poll = threading.Thread(target=self.monitor_process,
-                                        args=(iteration_done_event, idle_timeout_event))
+                                        args=(iteration_done_event, idle_timeout_event, monitor_launched))
                 poll.start()
+                assert monitor_launched.wait(30), "Failed to launch monitoring thread"
 
             def keep_waiting():
                 return self.target.monitor.is_healthy() and not idle_timeout_event.is_set()

